java/src/jmri/jmrit/display/layoutEditor/LayoutTrack.java

```diff
@@ -293,6 +293,33 @@ public String getTurnoutStateString(int turnoutState) {
         return result;
     }
 
+    /**
+     * Check for active block boundaries.
+     * <p>
+     * If any connection point of a track object has attached objects, such as signal
+     * masts, signal heads or NX sensors, the track component cannot be deleted.
+     * @return true if the track component can be deleted.
+     */
+    public abstract boolean canRemoveTrackComponent();
+
+    /**
+     * Display the attached items that prevent removing the component.
+     * @param itemList A list of the attached heads, masts and/or sensors.
+     * @param componentType The type such as Turnout, Level Crossing, etc.
+     */
+    public void displayRemoveTrackComponentDialog(List<String> itemList, String typeKey) {
+        itemList.sort(null);
+        StringBuilder msg = new StringBuilder(Bundle.getMessage("MakeLabel",  // NOI18N
+                Bundle.getMessage("DeleteTrackItem", Bundle.getMessage(typeKey))));  // NOI18N
+        for (String item : itemList) {
+            msg.append("\n    " + item);  // NOI18N
+        }
+        javax.swing.JOptionPane.showMessageDialog(null,
```

Comment:
Since the layout track objects are created in reference to a specific layoutEditor instance, shouldn’t this dialog be a child of that layoutEditor instance?  ( I.e. should the first barameter be something other than null? )

Reply:
Done.